pallets/ocw-testing/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config:
+        frame_system::offchain::SendTransactionTypes<Call<Self>> + frame_system::Config
+    {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// Number of blocks of cooldown after unsigned transaction is included.
+        ///
+        /// This ensures that we only accept unsigned transactions once, every `UnsignedInterval`
+        /// blocks.
+        #[pallet::constant]
+        type UnsignedInterval: Get<BlockNumberFor<Self>>;
+    }
+
+    #[pallet::storage]
+    pub(super) type OffchainWorkerTestEnabled<T> = StorageValue<_, bool, ValueQuery>;
+
+    /// Defines the block when next unsigned transaction will be accepted.
+    ///
+    /// To prevent spam of unsigned (and unpaid!) transactions on the network,
+    /// we only allow one transaction every `T::UnsignedInterval` blocks.
+    /// This storage entry defines when new transaction is going to be accepted.
+    #[pallet::storage]
+    pub(super) type NextUnsignedAt<T: Config> = StorageValue<_, BlockNumberFor<T>, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        pub _phantom_data: PhantomData<T>,
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            <OffchainWorkerTestEnabled<T>>::put(&false);
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        /// Offchain worker entry point.
+        ///
+        /// By implementing `fn offchain_worker` you declare a new offchain worker.
+        /// This function will be called when the node is fully synced and a new best block is
+        /// successfully imported.
+        /// Note that it's not guaranteed for offchain workers to run on EVERY block, there might
+        /// be cases where some blocks are skipped, or for some the worker runs twice (re-orgs),
+        /// so the code should be able to handle that.
+        fn offchain_worker(block_number: BlockNumberFor<T>) {
+            log::info!("Entering off-chain worker.");
+            // The entry point of your code called by off-chain worker
+            let res = Self::send_raw_unsigned_transaction(block_number);
+            if let Err(e) = res {
+                log::error!("Error: {}", e);
+            }
+        }
+    }
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Switches on or off the offchain worker
+        ///
+        /// Only root (or specified authority account) should be able to switch
+        /// the off-chain worker on and off to avoid enabling it by default in production
+        #[pallet::call_index(0)]
+        #[pallet::weight({0})]
+        pub fn switch_offchain_worker(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
Adding a bool parameter here will make this easier to use, `set` instead of `switch`

Reply:
Good point... updated it here https://github.com/moondance-labs/tanssi/pull/723/commits/e66e8733d74019b4df5d905fb63cf8faa3423d17